shakenfist/external_api/instance.py

```diff
@@ -501,114 +501,114 @@ def get(self, instance_uuid=None, instance_from_db=None):
 
 class InstanceEventsEndpoint(api_base.Resource):
     @jwt_required
-    @api_base.arg_is_instance_uuid
+    @api_base.arg_is_instance_ref
     @api_base.requires_instance_ownership
-    def get(self, instance_uuid=None, instance_from_db=None):
-        return list(db.get_events('instance', instance_uuid))
+    def get(self, instance_ref=None, instance_from_db=None):
+        return list(db.get_events('instance', instance_from_db.uuid))
 
 
 class InstanceRebootSoftEndpoint(api_base.Resource):
     @jwt_required
-    @api_base.arg_is_instance_uuid
+    @api_base.arg_is_instance_ref
     @api_base.requires_instance_ownership
     @api_base.redirect_instance_request
     @api_base.requires_instance_active
-    def post(self, instance_uuid=None, instance_from_db=None):
+    def post(self, instance_ref=None, instance_from_db=None):
         with db.get_lock(
-                'instance', None, instance_uuid, ttl=120, timeout=120,
+                'instance', None, instance_from_db.uuid, ttl=120, timeout=120,
```

Comment:
All of these `db.get_lock()` calls are a bit old school now. We should instead be using `instance_from_db.get_lock()`.

Reply:
To stay focused and minimise risk, I'll leave that change to the next PR. (We want to release this immediately.)